include/swift/AST/SimpleRequest.h

```diff
@@ -34,15 +34,33 @@ class Evaluator;
 /// Describes how the result for a particular request will be cached.
 enum class CacheKind {
   /// The result for a particular request should never be cached.
-  Uncached,
+  Uncached = 1 << 0,
   /// The result for a particular request should be cached within the
   /// evaluator itself.
-  Cached,
+  Cached = 1 << 1,
   /// The result of a particular request will be cached via some separate
   /// mechanism, such as a mutable data structure.
-  SeparatelyCached,
+  SeparatelyCached = 1 << 2,
+  /// This request introduces the source component of a source-sink
+  /// incremental dependency pair and defines a new dependency scope.
+  ///
+  /// This bit is optional.  High-level requests
+  /// (e.g. \c TypeCheckSourceFileRequest) will require it.
+  DependencySource = 1 << 3,
+  /// This request introduces the sink component of a source-sink
+  /// incremental dependency pair and is a consumer of the current
+  /// dependency scope.
+  ///
+  /// This bit is optional. Name lookup requests
+  /// (e.g. \c DirectLookupRequest) will require it.
+  DependencySink = 1 << 4,
```

Comment:
These comments are good, but in addition it would be good to add a cross-reference to the dependency document for the poor noob who has no idea what "dependency" means here.